### ISSUE TITLE
Progress meter for processing cache

### ISSUE THREAD

**Issue by janisozaur** [OpenRCT2/OpenRCT2#3190]
When you open a fresh game with many new objects, it takes time to process all the new items, during which time we don't present anything to the user and usually they assume something is not working.

We should have a progress bar stating we are still alive and busy processing things.

**Comment by RollingStar** [comment]
In vanilla RCT2, doesn't it pop up an OS-level progress bar?

**Comment by Richard-L** [comment]
It does indeed. Not super pretty but would do the job.

**Comment by Gymnasiast** [comment]
It's probably easier to create our own, actually, considering it would need code for all three OS's OpenRCT2 supports if we went down the OS-level route.

**Comment by PFCKrutonium** [comment]
Would we be able to just render a progress bar middle of the screen using SDL?

**Comment by janisozaur** [comment]
there shouldn't be much problem with that, at least from technical side.

**Comment by janisozaur** [comment]
Impact of that should be reduced by https://github.com/OpenRCT2/OpenRCT2/pull/3591 (see https://github.com/OpenRCT2/OpenRCT2/pull/3591#issuecomment-219174776) and https://github.com/OpenRCT2/OpenRCT2/pull/3592

**Comment by AaronVanGeffen** [comment]
This has since been implemented for the command line. Do we still want to implement this in the GUI, too?

**Comment by Broxzier** [comment]
Definitely. Most players don't see the console window, and have to wait 10+ seconds when the cache needs to rebuild.

**Comment by IntelOrca** [comment]
@AaronVanGeffen I have done it in a branch but I first I need to do a bit of refactoring so that I can use the window manager without having a park loaded.

**Comment by Liampie** [comment]
I'd like to bring this up to your attention again, the loading bars are sorely missed in my experience. Especially since loading the game tends to take 5-10 minutes for me.

**Comment by janisozaur** [comment]
I've never seen it load so slowly, even when pulling objects via network. What kind of setup do you have that causes such slowness?

Even better than adding progress bars would be speeding up the loading, perhaps via better caching, e.g. per object (mtime or file's sha1sum).

**Comment by IntelOrca** [comment]
@janisozaur I think he is referring to the building of the object cache which happens every time the object directory has new or less or changed objects. The time then is mostly down to the initial filesystem enumeration, file reading and sawyer decoding. The filesystem is OS specific and it is slow when the OS has not cached the directories but is the next time. It has improved a lot in Windows 10 when compared with XP.
